components/ledger/internal/machine/script/compiler/compiler.go

```diff
@@ -674,12 +684,14 @@ func CompileFull(input string) CompileArtifacts {
 	}
 
 	visitor := parseVisitor{
-		errListener:    errListener,
-		instructions:   make([]byte, 0),
-		resources:      make([]program.Resource, 0),
-		varIdx:         make(map[string]machine.Address),
-		neededBalances: make(map[machine.Address]map[machine.Address]struct{}),
-		sources:        map[machine.Address]struct{}{},
+		errListener:       errListener,
+		instructions:      make([]byte, 0),
+		resources:         make([]program.Resource, 0),
+		varIdx:            make(map[string]machine.Address),
+		neededBalances:    make(map[machine.Address]map[machine.Address]struct{}),
+		sources:           map[machine.Address]struct{}{},
+		writeLockAccounts: map[machine.Address]struct{}{},
+		readLockAccounts:  map[machine.Address]struct{}{},
```

Comment:
here the diff is weird because of golang's formatting, but the actual diff is this:
```diff
+ writeLockAccounts: map[machine.Address]struct{}{},
+ readLockAccounts:  map[machine.Address]struct{}{},
}
```